### PR TITLE
fixed issue with get prime factor number logic

### LINKED ISSUE #38286: get prime factors drops smaller factors when the largest prime factor exceeds sqrt(x)
### Description

Reproduces on jax 0.10.1 (latest release)

The `for … else` in `_get_prime_factors` (`jax/_src/mesh_utils.py`) has no `break`, so the `else` runs on loop completion and returns only the leftover prime, discarding the factors already seen. 

The loop trial divides up to `isqrt(x) + 1` (`range(2, isqrt(x) + 2)`), so any composite whose largest prime factor exceeds that bound returns an incorrect result.

```python
import math
from jax._src import mesh_utils
print(mesh_utils._get_prime_factors(10))                    # expected [2, 5], got [5]
print(math.prod(mesh_utils._get_prime_factors(10)) == 10)   # False
```

Same for 14 (`[2, 7]` → `[7]`), 22 (`[2, 11]` → `[11]`), 26 (`[2, 13]` → `[13]`), and so on

The bug stays hidden whenever every prime factor is ≤ `isqrt(x) + 1`, so `x` collapses to 1 inside the loop and takes the early `return`.

This covers 8 and 12, and also cases like 6 and 20, whose largest prime is exactly `isqrt(x) + 1` (note these are > `sqrt(x)` yet still handled correctly, so the bound is `isqrt(x) + 1`, not `sqrt(x)`).

Primes are fine too, since returning `[x]` is correct when no smaller factors were found.

**Downstream impact:** `_get_prime_factors` feeds `_enumerate_feasible_logical_axis_assignments`, so logical-mesh enumeration is wrong for any axis size with a large prime factor (10, 14, 22…).

Happy to open a PR with a fix and tests :)

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.10.1

jaxlib: 0.10.1

numpy:  2.4.6

python: 3.13.0 (v3.13.0:60403a5409f, Oct  7 2024, 00:37:40) [Clang 15.0.0 (clang-1500.3.9.4)]

device info: cpu-1, 1 local devices

process_count: 1

platform: uname_result(system='Darwin', node='MacBookPro.localdomain', release='25.5.0', machine='arm64')

```

### PR DESCRIPTION
**Pull request by leopozh**
Fixes https://github.com/jax-ml/jax/issues/38286

The for else in _get_prime_factors (jax/_src/mesh_utils.py) has no break, so the else runs on loop completion and returns only the leftover prime, discarding the factors already seen.

The loop trial divides up to isqrt(x) + 1 (range(2, isqrt(x) + 2)), so any composite whose largest prime factor exceeds that bound returns an incorrect result.

### CODE REVIEW

**Comment by yashk2810** [comment]
Thanks! Fixed in https://github.com/jax-ml/jax/commit/96e0bc7bcb4e14e063569e28d5f2606ec968c667
